src/Core/View.js

```diff
@@ -324,8 +341,29 @@ View.prototype.addLayer = function addLayer(layer, parentLayer) {
         this.scene.add(layer.object3d);
     }
 
-    this.notifyChange(true);
-    return layer.whenReady;
+    if (this._allLayersAreReadyCallback) {
+        // re-arm readyCallbacl
+        this.removeFrameRequester(MAIN_LOOP_EVENTS.AFTER_RENDER, this._allLayersAreReadyCallback);
+        this._allLayersAreReadyCallback = null;
+    }
+
+    return layer.whenReady.then((layer) => {
+        this.notifyChange(false);
+
+        if (!this._allLayersAreReadyCallback) {
+            this._allLayersAreReadyCallback = () => {
+                if (this.mainLoop.scheduler.commandsWaitingExecutionCount() == 0 &&
+                    this.mainLoop.renderingState == RENDERING_PAUSED) {
+                    this.dispatchEvent({ type: VIEW_EVENTS.LAYERS_INITIALIZED });
```

Comment:
in view of the dispatch conditions for event `LAYERS_INITIALIZED` also represents the loading of the data of the layers? and not only that the layer is ready

Reply:
Yes, LAYER_INITIALIZED mean: all currently added layers have loaded enough data to be displayable.

Reply:
could you complete the documentation on this event?

Reply:
Sure but be more specific. What should I add/remove/edit from [the existing doc](https://github.com/iTowns/itowns/pull/636/files#diff-d297dcf7b9d0e696ece4b1f6e1d7c69dR12)?

Reply:
the documentation doesn't explain the ready status and can be confused with `layer.whenReady`. This event indicates status of viewer, mainloop and the layers

Reply:
Done